### ISSUE TITLE
React Native Upgrade Breaking Tests

### ISSUE THREAD

**Issue by ReshmiCode** [wix/Detox#1532]
**Describe the bug**
After upgrading to RN 0.60.3 the tests just get stuck. The tests worked fine on version 0.59.9. The metro-bundler builds but the app is just idle and the test times out.

**To Reproduce**

- [x] I have tested this issue on the latest Detox release and it still reproduces

**Environment (please complete the following information):**
Detox: 12.8.1
React Native: 0.60.3
Node: 10.15.3
Device: iPhone 7
Xcode: 10.2.1
iOS: 12.2
macOS: 10.14.3

**Device and Verbose Detox Logs**
[errors.txt](https://github.com/wix/Detox/files/3392684/errors.txt)

**Comment by LeoNatan** [comment]
Thanks, I'll have a look. I don't think we've tested RN .60.3 yet.

**Comment by LeoNatan** [comment]
The test suite seems to run locally. Could you please follow the guide here and post back the log you are getting:
https://github.com/wix/Detox/blob/master/docs/Troubleshooting.Synchronization.md#lower-level-idling-resources-rebug-ios-only

**Comment by ReshmiCode** [comment]
@LeoNatan I added `launchArgs: { 'detoxPrintBusyIdleResources': 'YES' }` but when I run the command to obtain the log I get
`Error from getpwuid_r: 0 (Undefined error: 0)
Filtering the log data using "category == "EarlGreyStatistics""` and it just hangs

**Comment by LeoNatan** [comment]
That's odd. 

In your log, I see
```
detox[14367] TRACE: [DetoxServer.js/MESSAGE] role=tester action=isReady (sessionId=569d5a25-0d01-5c35-bfb3-6f981d5c84e4)
detox[14367] DEBUG: [DetoxServer.js/CANNOT_FORWARD] role=testee not connected, cannot fw action (sessionId=569d5a25-0d01-5c35-bfb3-6f981d5c84e4)
detox[14367] DEBUG: [DetoxServer.js/LOGIN] role=testee, sessionId=569d5a25-0d01-5c35-bfb3-6f981d5c84e4
detox[14367] DEBUG: [DetoxServer.js/LOGIN_SUCCESS] role=testee, sessionId=569d5a25-0d01-5c35-bfb3-6f981d5c84e4
detox[14367] DEBUG: [DetoxServer.js/DISCONNECT] role=testee, sessionId=569d5a25-0d01-5c35-bfb3-6f981d5c84e4
detox[14367] TRACE: [AsyncWebSocket.js/WEBSOCKET_MESSAGE] {"type":"testeeDisconnected","messageId":-49642}
```

It says that /testee/ (your app) connects to the detox server but then disconnects. I wonder, what is happening on the simulator while the test is hung. Is that last message printed before the timeout wait or after.

**Comment by ReshmiCode** [comment]
Before the timeout wait right when the app launches.

**Comment by LeoNatan** [comment]
If the app closes, that means it is crashing. Please follow this guide and post here what you see in Xcode once the crash happens:

https://github.com/wix/Detox/blob/master/docs/Guide.DebuggingInXcode.md

Thanks

**Comment by ReshmiCode** [comment]
Getting `Error: Could not find build script in detox.configurations["ios.none"].build
` when running the build command with -c ios.none. My config is
"ios.none": {
	"binaryPath": "ios",
	"type": "ios.none",
	"name": "iPhone 8 Plus",
	"session": {
		"server": "ws://localhost:8099",
		"sessionId": "guru.fan.app"
	}

**Comment by LeoNatan** [comment]
You are not supposed using Detox but using Xcode (from the UI).

**Comment by ReshmiCode** [comment]
Getting this
[xcode.txt](https://github.com/wix/Detox/files/3393530/xcode.txt)
Is it not connecting or something? The app doesn't open it just hangs on
detox[25147] DEBUG: [AsyncWebSocket.js/WEBSOCKET_OPEN] opened web socket to: ws://localhost:8099
detox[25147] TRACE: [AsyncWebSocket.js/WEBSOCKET_SEND] {"type":"login","params":{"sessionId":"guru.fan.app","role":"tester"},"messageId":0}
detox[25147] TRACE: [AsyncWebSocket.js/WEBSOCKET_MESSAGE] {"type":"loginSuccess","params":{"sessionId":"guru.fan.app","role":"tester"},"messageId":0}
detox[25147] TRACE: [AsyncWebSocket.js/WEBSOCKET_SEND] {"type":"isReady","params":{},"messageId":-1000}

**Comment by LeoNatan** [comment]
That log doesn't include Detox messages. Please follow the guide.

**Comment by ReshmiCode** [comment]
I've followed it. I set this up:
<img width="1204" alt="Screen Shot 2019-07-16 at 8 59 12 AM" src="https://user-images.githubusercontent.com/16989366/61300744-2ba3a500-a7a8-11e9-9380-850f0a9ba1bd.png">
Then ran the server, built the app as I would normally do to an iPhone 7 with the run button, then ran the tests.
![Screen Shot 2019-07-16 at 9 02 06 AM](https://user-images.githubusercontent.com/16989366/61300919-85a46a80-a7a8-11e9-9677-a5a36ee1360a.png)

**Comment by ReshmiCode** [comment]
Just found this error
dyld: Library not loaded: @rpath/COSTouchVisualizer.framework/COSTouchVisualizer
  Referenced from: /private/var/containers/Bundle/Application/D76AEF94-EE01-4F5C-AFF5-51D061A0891A/Fanguru.app/Frameworks/Detox.framework/Detox
  Reason: no suitable image found.  Did find:
	/private/var/containers/Bundle/Application/D76AEF94-EE01-4F5C-AFF5-51D061A0891A/Fanguru.app/Frameworks/Detox.framework/Frameworks/COSTouchVisualizer.framework/COSTouchVisualizer: code signature in (/private/var/containers/Bundle/Application/D76AEF94-EE01-4F5C-AFF5-51D061A0891A/Fanguru.app/Frameworks/Detox.framework/Frameworks/COSTouchVisualizer.framework/COSTouchVisualizer) not valid for use in process using Library Validation: mapped file has no cdhash, completely unsigned? Code has to be at least ad-hoc signed.

**Comment by LeoNatan** [comment]
That's quite odd. Not sure what is going on.

**Comment by ReshmiCode** [comment]
The app that builds doesn't work it just closes and the buttons don't work. If I remove the embedded binary the app works fine.
Is this the right config for it?
<img width="1373" alt="Screen Shot 2019-07-16 at 12 54 47 PM" src="https://user-images.githubusercontent.com/16989366/61317773-4c302700-a7c9-11e9-8cc2-f1199dc695cf.png">

**Comment by LeoNatan** [comment]
Yes, that is the way to configure. 

Let's try something different. Please clone Detox from source (this repo) **recursively**, then instead of integrating the Detox.xcodeproj from your node modules, integrate the project from `<clone dir>/detox/ios` and try to run your project.

**Comment by ReshmiCode** [comment]
Do you mean add the embedded binary `Detox.framework` from the cloned directory's `Detox.xcodeproj`?

**Comment by LeoNatan** [comment]
Yes. Remove the Detox project you previously added, add the one from the cloned dir, and finally add it to embedded binaries. Now try to build your project.

Just make sure you recursively clone or update submodules recursively after cloning.

**Comment by ReshmiCode** [comment]
Tried that but the built app still doesn't function properly. The build logs are different. With the Detox binary:
![Screen Shot 2019-07-16 at 3 50 05 PM](https://user-images.githubusercontent.com/16989366/61328867-945b4380-a7e1-11e9-8803-4e1298668177.png)
Without:
![Screen Shot 2019-07-16 at 3 50 29 PM](https://user-images.githubusercontent.com/16989366/61328874-97563400-a7e1-11e9-9bed-87722c578109.png)
Both builds succeed but with the binary the app looses all functionality.

**Comment by LeoNatan** [comment]
I am sorry, I don't understand. What does "looses all functionality" mean? Is the app stuck?

**Comment by LeoNatan** [comment]
This has turned into Xcode support, which I don't like here. We normally direct people to Stack Overflow.
Please provide a demo project which reproduces this issue or consult with Stack Overflow how to debug this issue further.

**Comment by ReshmiCode** [comment]
The app just shows the start screen without the buttons work or api calls being made. This issue goes away as soon as I remove the Detox embedded binary.
I can try to figure out how to get the logs by myself but looks like something in the docs is outdated (https://github.com/wix/Detox/blob/master/docs/Guide.DebuggingInXcode.md).

**Comment by ReshmiCode** [comment]
The Detox embedded binary seems to be the issue.
<img width="1169" alt="Screen Shot 2019-07-17 at 11 06 15 AM" src="https://user-images.githubusercontent.com/16989366/61397641-f4a8be80-a88f-11e9-9aaf-fb4b94d7074c.png">

On an old working project with Detox, https://github.com/ReshmiCode/Detox-Test following the same steps just results in the build crashing sooner with errors like
`:-1: /Users/QAintern/Desktop/Detox/Detox.xcconfig: unable to open file (in project "Detox") (in target 'Detox')
:-1: Unable to resolve build file: XCBCore.BuildFile (namedReferencesCannotBeResolved) (in target 'Detox')`

**Comment by LeoNatan** [comment]
It's something on your machine. Detox, with that embedded framework, work correctly for everyone else. Not sure why signature is breaking for you.

**Comment by ReshmiCode** [comment]
I had a colleague try it on their computer with our project and the same error occurred. Are there any logs on Android that could be useful? Fixed the Android build and the same problem occurs with the tests.

**Comment by LeoNatan** [comment]
If you have the same problem on Android as iOS, it's a 99.999999% problem in your project. I am less familiar with Android Detox. @d4vidi @rotemmiz Any logs on Android to help with sync issues?

**Comment by ReshmiCode** [comment]
In the process of making a video to showcase the error, the Android tests started working. But I noticed that when the tests are normally run (detox build then detox test), the iOS version looses functionality the same way as when adding the embedded binary. https://www.loom.com/share/520cb8e01fd24adabc5f2bcf7fd033aa
If I open the build by myself the app works but if it opens through the detox test command the app's buttons don't work (as shown in the video at the end).

**Comment by LeoNatan** [comment]
When you npm install for the first time, Detox is built from source. It makes sense that it will fail the same way as it does when embedded in the project. But this is unrelated to RN version as you claimed originally.

**Comment by ReshmiCode** [comment]
Yeah sorry about that, I assumed it was due to the RN version and until Android worked I assumed it was. The video is taken from when Detox was launched normally (through detox build).

**Comment by LeoNatan** [comment]
What is shown in the video is the app being launched. That means that it has successfully loaded Detox. So the problem there is different.
OK, in your `~/Library/Detox` there should be one or more folders. If one, proceed to step 2.
#### Step 1 (for convenience)
Delete all folders in `~/Library/Detox`
Delete your node_modules and `npm install`—this will generate a new folder under `~/Library/Detox`

#### Step 2
Don't follow the guide—if you have, remove Detox from your project
Instead, drag the Detox.framework from `~/Library/Detox/{...}` into your Xcode project
Add the newly added framework into **Embedded Binaries** (like it shows in the guide)

#### Step 3
Add an Objective C exception breakpoint before running (Google how to do this)
Run your project from Xcode
Start the test
Once your project crashes (which we suspect it does), head into the bottom section in Xcode, where there should be a console output and you can type things
Type `bt` and press return
Copy the entire log, post it to https://gist.github.com/ and link here

Thanks

**Comment by ReshmiCode** [comment]
https://www.loom.com/share/d6283756c8e3468695a0e87ee42e3f38
The beginning of the video shows that without the embedded the app works but after adding it, the buttons don't work. The breakpoint wasn't hit so I believe I couldn't type bt in the log section.

**Comment by LeoNatan** [comment]
Oddly, you don't have the pause button. Perhaps debugging is disabled in the scheme? In the scheme editor for your app, make sure "Debug executable" is enabled.

**Comment by ReshmiCode** [comment]
Here's the log
https://gist.github.com/ReshmiCode/1d14943ecbf9a35db4c56ec104919018
I'm getting this after I build the app, before I can run the tests.

**Comment by LeoNatan** [comment]
Great, this has been solved in 13.2.0. Please upgrade.
